lib/tds.js

```diff
@@ -290,19 +309,30 @@ async function generateDNRRulesForTrackerEntry (
         }
 
         priority += TRACKER_RULE_PRIORITY_INCREMENT
-        dnrRules.push(
-            generateDNRRule({
-                priority,
-                actionType: ruleAction,
-                redirect: redirectAction,
-                urlFilter,
-                regexFilter,
-                matchCase,
-                requestDomains,
-                excludedInitiatorDomains,
-                resourceTypes
-            })
-        )
+
+        const newRule = generateDNRRule({
+            priority,
+            actionType: ruleAction,
+            redirect: redirectAction,
+            urlFilter,
+            regexFilter,
+            matchCase,
+            requestDomains,
+            excludedInitiatorDomains,
+            resourceTypes
+        })
+
+        if (customRuleAction) {
+            // if a custom action, map the action to the specified default
+            // and add its inverse to the list of custom action rules
+            const inverseRule = { customAction: trackerEntryRules[i].action, ...newRule }
```

Comment:
I would prefer to call `generateDNRRule` again here, but with the inverse action. Otherwise, we're sharing some Objects between the two rules which could lead to confusing bugs where one is mutated and the other rule changes in an unexpected way.

Reply:
Done

Reply:
Done!

Reply:
Thanks, yep either would work. To be fair, that would probably make more sense too (see below)!